tests/FakeItEasy.IntegrationTests.VB/RaisingEventsTests.vb

```diff
@@ -176,4 +176,21 @@ Public Class RaisingEventsTests
         ' Assert
         ReferenceEquals(capturedObject, anObject).Should().BeTrue()
     End Sub
+
+    <Fact>
+    <SuppressMessage("Microsoft.Design", "CA1030:UseEventsWhereAppropriate",
+        Justification:="Required for testing.")>
+    Public Sub RaiseFreeform_ObjectEvent_sends_object()
+        'Arrange
+        Dim target = A.Fake(Of IHaveEvents)()
+        Dim anObject As Object = New Object()
+
+        AddHandler target.ObjectEvent, AddressOf HandlesObjectEvent
+
+        ' Act
+        AddHandler target.ObjectEvent, Raise.FreeForm(Of IHaveEvents.ObjectEventEventHandler).With(anObject)
+
+        ' Assert
+        ReferenceEquals(capturedObject, anObject).Should().BeTrue()

```

Comment:
Nitpick: could be written as `capturedObject.Should().BeSameAs(anObject)`

Reply:
'''

Reply:
I think that's why we were using `ReferenceEquals` to begin with.  I can comment up, if you like.

Reply:
That's weird... there's no late binding involves here, as far as I can tell

Reply:
Apparently VB.NET doesn't support extension methods on `Object`; this is related to the fact that there is no `dynamic` in VB.NET : anything typed as `Object` is late bound.

More details here: https://blogs.msdn.microsoft.com/vbteam/2007/01/24/extension-methods-and-late-binding-extension-methods-part-4/

Reply:
Wow. Thanks for the note. If you'd like, I can introduce a method to help us document what's going on here. Or change `Object` to something else. It doesn't _need_ to be `Object`.

Reply:
Not sure it's worth the trouble... Maybe changing to something other than `Object` would be enough.

Reply:
Did so. But then noticed the "sender", which is an object. I pushed a thing. Hate it, if you want!

Reply:
> Did so. But then noticed the "sender", which is an object. I pushed a thing. Hate it, if you want!

Perfect!